core/arch/arm/mm/tee_pager.c

```diff
@@ -484,6 +485,21 @@ static void tee_pager_load_page(struct tee_pager_area *area, vaddr_t page_va,
 {
 	size_t idx = (page_va - area->base) >> SMALL_PAGE_SHIFT;
 	const void *stored_page = area->store + idx * SMALL_PAGE_SIZE;
+	struct core_mmu_table_info *ti __maybe_unused;
+	uint32_t attr_alias __maybe_unused;
+	paddr_t pa_alias __maybe_unused;
+	unsigned int idx_alias __maybe_unused;
+
+	/* Insure we are allowed to write to aliased virtual page */
+	ti = &pager_alias_tbl_info;
+	idx_alias = core_mmu_va2idx(ti, (vaddr_t)va_alias);
+	core_mmu_get_entry(ti, idx_alias, &pa_alias, &attr_alias);
+	if (!(attr_alias & TEE_MATTR_PW)) {
+		attr_alias |= TEE_MATTR_PW;
+		core_mmu_set_entry(ti, idx_alias, pa_alias, attr_alias);
+		/* TODO: flush TLB for target page only */
+		core_tlb_maintenance(TLBINV_UNIFIEDTLB, 0);
```

Comment:
If we always made sure that aliased pages aren't accessible unless in this function there wouldn't be any need for this flush. But that's something for another PR.